CRM/Activity/BAO/Activity.php

```diff
@@ -1254,6 +1265,48 @@ public static function sendEmail(
     return [$sent, $activityID];
   }
 
+  /**
+   * Returns a array of attachment key with matching file ID.
+   *
+   * The function searches for all file Ids added for the activity and returns an array that
+   * uses the attachment key as the key and the file ID in the database for that matching attachment
+   * key by comparing the file URI for that attachment to the matching file URI fetched from the
+   * database. Having the file id matched per attachment key helps not to create a new file entry
+   * when a new activity with these attachments when the email activity is created.
+   *
+   * @param int $activityID
+   *   Activity Id.
+   * @param array $attachments
+   *   Attachments.
+   *
+   * @return array
+   *   Array of attachment key versus file Id.
+   */
+  private static function getAttachmentFileIds($activityID, $attachments) {
+    $queryParams = [1 => [$activityID, 'Positive'], 2 => [CRM_Activity_DAO_Activity::getTableName(), 'String']];
```

Comment:
I think the attachment api would do this in an ACL'd way - is that right @totten

Reply:
Does ACL apply here? If it would apply to the logged in user, it's their own file they're uploading, and they'd already have whatever permissions needed for activity creation. If for the recipient, is that doable through ACLs? - that would be an interesting feature.